py/desisim/scripts/quickquasars.py

```diff
@@ -0,0 +1,220 @@
+                log.error("Cannot guess nside and healpix from filename {}".format(ifilename))
+                raise ValueError("Cannot guess nside and healpix from filename {}".format(ifilename))
+            try :
+                healpix=int(vals[-1])
+                nside=int(vals[-2])
+            except ValueError:
+                raise ValueError("Cannot guess nside and healpix from filename {}".format(ifilename))
+            
+            ofilename = os.path.join(args.outdir,"{}/{}/spectra-{}-{}.fits".format(healpix//100,healpix,nside,healpix))
+        
+        if not args.overwrite :
+            if os.path.isfile(ofilename) :
+                log.info("skip existing {}".format(ofilename))
+                continue
+
+        log.info("Read skewers in {}".format(ifilename))
+        trans_wave, transmission, metadata = read_lya_skewers(ifilename)
+        ok = np.where(( metadata['Z'] >= args.zmin ) & (metadata['Z'] <= args.zmax ))[0]
+        transmission = transmission[ok]
+        metadata = metadata[:][ok]
+
+
+
+        # create quasars
+        nqso=transmission.shape[0]
+        if args.downsampling is not None :
+            if args.downsampling <= 0 or  args.downsampling > 1 :
+               log.error("Down sampling fraction={} must be between 0 and 1".format(args.downsampling))
+               raise ValueError("Down sampling fraction={} must be between 0 and 1".format(args.downsampling))
+            indices = np.where(np.random.uniform(size=nqso)<args.downsampling)[0]
+            if indices.size == 0 : 
+                log.warning("Down sampling from {} to 0 (by chance I presume)".format(nqso))
+                continue
+            transmission = transmission[indices]
+            metadata = metadata[:][indices]
+            nqso = transmission.shape[0]
+
+        if args.nmax is not None :
+            if args.nmax < nqso :
+                log.info("Limit number of QSOs from {} to nmax={} (random subsample)".format(nqso,args.nmax))
+                # take a random subsample
+                indices = (np.random.uniform(size=args.nmax)*nqso).astype(int)
+                transmission = transmission[indices]
+                metadata = metadata[:][indices]
+                nqso = args.nmax
+
+        log.info("Simulate {} QSOs".format(nqso))
+        tmp_qso_flux, tmp_qso_wave, meta = model.make_templates(
+            nmodel=nqso, redshift=metadata['Z'], seed=args.seed,
+            lyaforest=False, nocolorcuts=True, noresample=True)
+
+        log.info("Resample to a linear wavelength grid (needed by DESI sim.)")
+        qso_wave=np.linspace(args.wmin,args.wmax,int((args.wmax-args.wmin)/args.dwave)+1)
+        qso_flux=np.zeros((tmp_qso_flux.shape[0],qso_wave.size))
+        for q in range(tmp_qso_flux.shape[0]) :
+            qso_flux[q]=np.interp(qso_wave,tmp_qso_wave,tmp_qso_flux[q])
+
+
+        log.info("Apply lya")
+        qso_flux = apply_lya_transmission(qso_wave,qso_flux,trans_wave,transmission)
+
```

Comment:
If you want to apply target selection you can do it here with

```
from desitarget.cuts import isQSO_colors
isqso = isQSO_colors(gflux=meta['FLUX_G'], rflux=meta['FLUX_R'], 
    zflux=meta['FLUX_Z'], w1flux=meta['FLUX_W1'], w2flux=meta['FLUX_W2'])
```
and just keep targets with `isqso == 1`.